src/panels/config/automation/condition/types/ha-automation-condition-state.ts

```diff
@@ -22,6 +23,25 @@ export class HaStateCondition extends LitElement implements ConditionElement {
     return { entity_id: "", state: "" };
   }
 
+  protected shouldUpdate(changedProperties: PropertyValues): boolean {
+    if (
+      changedProperties.has("condition") &&
+      this._stateIsArray(this.condition)
+    ) {
+      if (this._stateIsArray(this.condition)) {
+        fireEvent(
+          this,
+          "ui-mode-not-available",
+          Error(this.hass.localize("ui.errors.config.editor_not_supported"))
```

Comment:
Should I add another more specific error here? Maybe something like "State arrays not supported in the visual editor"

Reply:
Yes, I think we should